packages/webpack/src/plugins/nx-webpack-plugin/lib/apply-base-config.ts

```diff
@@ -230,6 +234,41 @@ function applyNxDependentConfig(
     root: options.root,
   };
 
+  if (!options.buildLibsFromSource && executorContext.targetName) {
+    const remappedTarget =
+      executorContext.targetName === 'serve'
+        ? 'build'
+        : executorContext.targetName;
+
+    const { target, dependencies } = calculateProjectBuildableDependencies(
+      undefined,
+      executorContext.projectGraph,
+      executorContext.root,
+      executorContext.projectName,
+      remappedTarget,
+      executorContext.configurationName
+    );
+    options.tsConfig = createTmpTsConfig(
+      tsConfig,
+      executorContext.root,
+      target.data.root,
+      dependencies
+    );
+    tsConfig = options.tsConfig;
+
+    if (executorContext.targetName === 'serve') {
+      const buildableDependencies = dependencies
+        .filter((dependency) => dependency.node.type === 'lib')
+        .map((dependency) => dependency.node.name)
+        .join(',');
+
+      const buildCommand = `nx run-many --target=build --projects=${buildableDependencies}`;
+
+      plugins.push(new WebpackNxBuildCoordinationPlugin(buildCommand));
+    }
+  }
+  plugins.push(new NxTsconfigPathsWebpackPlugin({ tsConfig }));
```

Comment:
Can we move this to the `NxTsconfigPathsWebpackPlugin`?
It would prevent adding this plugin twice and is a more correct location for the logic